pyhf/readxml.py

```diff
@@ -26,13 +26,18 @@ def extract_error(h):
     return np.sqrt(err).tolist()
 
 
-def import_root_histogram(rootdir, filename, path, name):
+def import_root_histogram(rootdir, filename, path, name, filecache={}):
```

Comment:
yes if I'm not mistaken python can be a bit annoying with having mutable objects as default arguments

```pyhton
__FILECACHE__ = []

def import_root_histogram(rootdir, filename, path, name, filecache=None):
    global __FILECACHE__
    filecache = filecache or __FILECACHE__
```

should do the trick

Reply:
In this case the python behavior (dictionary is created when function is defined) was intended, so the function has a persistent cache during multiple executions. But i agree that a global file cache in the module is probably better. The question remains where/if to automatically delete the cache?

Reply:
we could have a `readxml.clearcache()` method . but honestly I don't think we will run into this often.. I expect that most of the time the readxml module will be called from the command line via `pyhf xml2json` and the cache will be torn out on exit anyways.. we can add this later on if this becomes a problem

Reply:
(unless you already ran into a situation where this takes up too much memory)

Reply:
> but honestly I don't think we will run into this often.. 

.... true - if the the file was produced with HistFitter it is anyways just one root file